ledger/src/shred.rs

```diff
@@ -306,24 +277,13 @@ impl Shred {
             data_header.flags |= ShredFlags::LAST_SHRED_IN_SLOT
         }
 
-        let mut start = 0;
-        Self::serialize_obj_into(
-            &mut start,
-            SIZE_OF_COMMON_SHRED_HEADER,
-            &mut payload,
-            &common_header,
-        )
-        .expect("Failed to write common header into shred buffer");
-
-        Self::serialize_obj_into(
-            &mut start,
-            SIZE_OF_DATA_SHRED_HEADER,
-            &mut payload,
-            &data_header,
-        )
-        .expect("Failed to write data header into shred buffer");
+        let mut cursor = Cursor::new(&mut payload[..]);
+        bincode::serialize_into(&mut cursor, &common_header).unwrap();
+        bincode::serialize_into(&mut cursor, &data_header).unwrap();
         // TODO: Need to check if data is too large!
-        payload[start..start + data.len()].copy_from_slice(data);
+        let offset = cursor.position() as usize;
+        debug_assert_eq!(offset, SHRED_DATA_OFFSET);
```

Comment:
What do you think about moving this into a unit test (like `test_shred_constants`) ?

Reply:
There is already a lot of test coverage over this line.
Also https://github.com/solana-labs/solana/pull/24866 added tests for backward compatibility of shreds serialization.
if you for example change this to `offset+1` you will see a lot of tests will fail.

Reply:
Sure. although coverage of this constant is indirect as opposed to the other constants where their values are directly asserted against `bincode::serialized_size()` of that struct. My point was more so that this is essentially testing the following
```
SHRED_DATA_OFFSET == serialized_size(ShredCommonHeader) + serialized_size(DataShredHeader)
```
The right hand side is constant at runtime (will only change if the struct layout changes), so it seemed unnecessary to have this check at runtime every-time the function is called (although maybe that is why you made it `debug_assert!()` instead of regular `assert!()` such that we don't incur any cost in release binary